### ISSUE TITLE
implement issues_by_query

### ISSUE THREAD

**Issue by kjaymiller** [Python-Community-News/GH-Issues#29]
#### TLDR

implement the function 

```py
def issues_by_query(query:str, api_key: str|None = None) -> Iterator[Issue]:
  """
  Perform a GitHub query for issues:
   - iterate through the results returning an Issue object for each one
  """
```

##### How to handle API Rate Limits?

I think the easiest answer is to accept an `api_key`.

- [ ] [QUESTION] Should we implement better Errors or pass-through the error from GitHub.

#### Considerations

We wanted to create an intuitive way to look at a list of queries

We considered a query object that users could apply each object

```python 
Query("Django") \ Preform Bulk Search
    .with_label("conferences") \  filter for label
    .from_org("Blackpythondevs") \ filter for label
    .from_repo("Blackpythondevs.github.io")

Query.search()
```

Ultimately, this would need to still call some function doing the work. It would also require more testing and documentation.

Even if you have a bunch of attributes that you can pass in directly. It doesn't solve the issue.

```python
Query("Django", labels=["confernces"], repo="blackpythondevs/Blackpythondevs")
```

There was also the concern about some of the query syntax we'd need to account for like OR|AND .

```python
    def to_string(attr:str, value:str):
        return f"{attr}:{value}"

    def _or(*args):
        "OR ".join(args)

    def _and(*args):
        "AND ".join(arg)

    def multi_strings
```

There was also some conversation about creating some syntactic sugar, but ultimately it would all come down to a function so let's focus on the function.
